### PR TITLE
Adds: promote cli command.

### PR DESCRIPTION
**Pull request by korvyashka**
## What does this PR do?
- Adds cli command edge app promote 
```
screenly edge-app promote --help        
Usage: screenly edge-app promote [OPTIONS] [PATH]

Arguments:
  [PATH]  Path to the directory with the manifest. If not specified CLI will use the current working directory

Options:
  -r, --revision <REVISION>  Edge app revision to promote
  -h, --help                 Print help
  -V, --version              Print version
```

- Adds optional commented lines for local development(with local screenly pro)
## GitHub issue or Phabricator ticket number?
https://ph.wireload.net/T7147

## How has this been tested?
Manuall on local environment and unit tests

## Checklist before merging

- [x] If have added tests.
- [x] Is this a new feature? If yes, please write one phrase about this update.
- [ ] I've attached relevant screenshots (if relevant).

### CODE REVIEW

File: src/authentication.rs [hunk 1]
```diff
@@ -114,7 +118,9 @@ fn verify_token(token: &str, api_url: &str) -> anyhow::Result<(), Authentication
     // Using uuid of non existing playlist. If we get 404 it means we authenticated successfully.
     let url = format!("{}/v3/groups/11CF9Z3GZR0005XXKH00F8V20R/", api_url);
     let secret = format!("Token {token}");
-    let client = reqwest::blocking::Client::builder().build()?;
+    let client = reqwest::blocking::Client::builder()
+        //.danger_accept_invalid_certs(true)
```
Comment (sergey-borovkov) [inline src/authentication.rs:122]: same

File: src/commands/edge_app.rs [hunk 2]
```diff
@@ -185,6 +185,37 @@ impl EdgeAppCommand {
         Ok(())
     }
 
+    pub fn promote(&self, manifest: &EdgeAppManifest, version: &Option<i32>) -> Result<i32, CommandError> {
```
Comment (sergey-borovkov) [inline src/commands/edge_app.rs:188]: just pass manifest - manifest has revision. If user sets version in the cli, modify manifest struct

File: src/commands/edge_app.rs [hunk 3]
```diff
@@ -185,6 +185,37 @@ impl EdgeAppCommand {
...
+                }
+            ),
+        };
+        let response = commands::post(
+            &self.authentication,
+            "v4/edge-apps/promote",
+            &payload,
+        )?;
+
+        if let Some(dict) = response.as_object() {
```
Comment (sergey-borovkov) [inline src/commands/edge_app.rs:208]: you can reduce nesting here by transforming it in something like
let revision_value = response
    .as_object()
    .and_then(|dict| dict.get("updated"))
    .and_then(|revision| revision.as_i64())
    .map(|revision_value| revision_value as i32);

File: src/authentication.rs [hunk 4]
```diff
@@ -89,6 +92,7 @@ impl Authentication {
         );
 
         reqwest::blocking::Client::builder()
+            //.danger_accept_invalid_certs(true)
```
[the remark this answers is not in the document]
Comment (korvyashka) [reply src/authentication.rs:95]: That is the same as commented stage API - if you need to do a local dev development, you would need to uncomment this as I explained in comment near the API.

File: src/cli.rs [hunk 5]
```diff
@@ -766,6 +774,22 @@ pub fn handle_cli_edge_app_command(command: &EdgeAppCommands) {
...
+        EdgeAppCommands::Promote{path, revision} => {
+            if let Some(_revision) = revision {
+                println!("Revision not specified. Latest revision will be used to promote.");
+            }
+            match edge_app_command.promote(
+                &EdgeAppManifest::new(transform_edge_app_path_to_manifest(path).as_path()).unwrap(),
+                revision
+            ) {
+                Ok(updated) => {
+                    println!("Edge app successfully promoted. Updated playlist items: {updated}.");
```
[the remark this answers is not in the document]
Comment (korvyashka) [reply src/cli.rs:786]: It prints amount of the updated playlist items - I think it is informative.

File: src/commands/edge_app.rs (hunk 2 above)
Comment (korvyashka) [reply src/commands/edge_app.rs:188]: This version could be different than in the manifest - for example downgrading to some previous version.

**Review by sergey-borovkov — APPROVED** [review]

**Pull request merged**